### ISSUE TITLE
undefined method `call' for nil:NilClass

### ISSUE THREAD

**Issue by farmanp** [neo4jrb/neo4j#1335]
I'm trying to run neo4j on a rails app in development and I followed all the steps that were written here http://neo4jrb.readthedocs.io/en/8.0.x/Setup.html but for some reason, after I set options in the application.rb file for a session, I get this error:

undefined method `call' for nil:NilClass

Additional information which could be helpful if relevent to your issue:

### Code example (inline, gist, or repo)
end
      Neo4j::Core::CypherSession::Adaptors::Base.subscribe_to_query(&logger_proc)
      TYPE_SUBSCRIBERS[session_type.to_sym].call(&logger_proc)

      @neo4j_cypher_logging_registered = true
    end

### Runtime information:
NoMethodError (undefined method `call' for nil:NilClass):
  neo4j (8.0.2) lib/neo4j/railtie.rb:139:in `register_neo4j_cypher_logging'
  neo4j (8.0.2) lib/neo4j/railtie.rb:66:in `setup!'
  neo4j (8.0.2) lib/neo4j/railtie.rb:55:in `block (2 levels) in <class:Railtie>'
  neo4j (8.0.2) lib/neo4j/active_base.rb:18:in `call'
  neo4j (8.0.2) lib/neo4j/active_base.rb:18:in `establish_session'
  neo4j (8.0.2) lib/neo4j/active_base.rb:8:in `current_session'
  neo4j (8.0.2) lib/neo4j/shared.rb:17:in `neo4j_session'
  neo4j (8.0.2) lib/neo4j/active_node/labels.rb:143:in `mapped_label'
  neo4j (8.0.2) lib/neo4j/migrations/runner.rb:16:in `initialize'
  neo4j (8.0.2) lib/neo4j/migrations.rb:13:in `new'
  neo4j (8.0.2) lib/neo4j/migrations.rb:13:in `check_for_pending_migrations!'
  neo4j (8.0.2) lib/neo4j/migrations/check_pending.rb:13:in `call'
  actionpack (4.2.5.1) lib/action_dispatch/middleware/callbacks.rb:29:in `block in call'
  activesupport (4.2.5.1) lib/active_support/callbacks.rb:88:in `__run_callbacks__'
  activesupport (4.2.5.1) lib/active_support/callbacks.rb:778:in `_run_call_callbacks'
  activesupport (4.2.5.1) lib/active_support/callbacks.rb:81:in `run_callbacks'
  actionpack (4.2.5.1) lib/action_dispatch/middleware/callbacks.rb:27:in `call'
  actionpack (4.2.5.1) lib/action_dispatch/middleware/reloader.rb:73:in `call'
  actionpack (4.2.5.1) lib/action_dispatch/middleware/remote_ip.rb:78:in `call'
  actionpack (4.2.5.1) lib/action_dispatch/middleware/debug_exceptions.rb:17:in `call'
  web-console (2.3.0) lib/web_console/middleware.rb:28:in `block in call'
  web-console (2.3.0) lib/web_console/middleware.rb:18:in `catch'
  web-console (2.3.0) lib/web_console/middleware.rb:18:in `call'
  actionpack (4.2.5.1) lib/action_dispatch/middleware/show_exceptions.rb:30:in `call'
  railties (4.2.5.1) lib/rails/rack/logger.rb:38:in `call_app'
  railties (4.2.5.1) lib/rails/rack/logger.rb:20:in `block in call'
  activesupport (4.2.5.1) lib/active_support/tagged_logging.rb:68:in `block in tagged'
  activesupport (4.2.5.1) lib/active_support/tagged_logging.rb:26:in `tagged'
  activesupport (4.2.5.1) lib/active_support/tagged_logging.rb:68:in `tagged'
  railties (4.2.5.1) lib/rails/rack/logger.rb:20:in `call'
  actionpack (4.2.5.1) lib/action_dispatch/middleware/request_id.rb:21:in `call'
  rack (1.6.5) lib/rack/methodoverride.rb:22:in `call'
  rack (1.6.5) lib/rack/runtime.rb:18:in `call'
  activesupport (4.2.5.1) lib/active_support/cache/strategy/local_cache_middleware.rb:28:in `call'
  rack (1.6.5) lib/rack/lock.rb:17:in `call'
  actionpack (4.2.5.1) lib/action_dispatch/middleware/static.rb:116:in `call'
  rack (1.6.5) lib/rack/sendfile.rb:113:in `call'
  railties (4.2.5.1) lib/rails/engine.rb:518:in `call'
  railties (4.2.5.1) lib/rails/application.rb:165:in `call'
  rack (1.6.5) lib/rack/lock.rb:17:in `call'
  rack (1.6.5) lib/rack/content_length.rb:15:in `call'
  rack (1.6.5) lib/rack/handler/webrick.rb:88:in `service'
  /Users/farmanp/.rbenv/versions/2.2.1/lib/ruby/2.2.0/webrick/httpserver.rb:138:in `service'
  /Users/farmanp/.rbenv/versions/2.2.1/lib/ruby/2.2.0/webrick/httpserver.rb:94:in `run'
  /Users/farmanp/.rbenv/versions/2.2.1/lib/ruby/2.2.0/webrick/server.rb:294:in `block in start_thread'

`neo4j` gem version: 8.0.0

**Comment by cheerfulstoic** [comment]
Have you updated to the latest versions of the gems (8.0.0 for `neo4j` and `7.0.2` for `neo4j-core`)?  This might already be fixed

**Comment by Joshfindit** [comment]
Did a bundle update from `neo4j-core 6.1.6 `, `neo4j 7.2.3`, and `rails 5.0.1.rc1` to `neo4j-core 7.0.2`, `neo4j 8.0.2`, and `rails 5.0.1`.

Now getting the same error when doing `Model.all` (specifically, `render Model.all` and `Model.all.first`):

```
irb(main):002:0> Artefact.all.first
NoMethodError: undefined method `call' for nil:NilClass
	from /usr/local/bundle/gems/neo4j-8.0.2/lib/neo4j/railtie.rb:139:in `register_neo4j_cypher_logging'
	from /usr/local/bundle/gems/neo4j-8.0.2/lib/neo4j/railtie.rb:66:in `setup!'
	from /usr/local/bundle/gems/neo4j-8.0.2/lib/neo4j/railtie.rb:55:in `block (2 levels) in <class:Railtie>'
	from /usr/local/bundle/gems/neo4j-8.0.2/lib/neo4j/active_base.rb:18:in `establish_session'
	from /usr/local/bundle/gems/neo4j-8.0.2/lib/neo4j/active_base.rb:8:in `current_session'
	from /usr/local/bundle/gems/neo4j-8.0.2/lib/neo4j/model_schema.rb:35:in `model_constraints'
	from /usr/local/bundle/gems/neo4j-8.0.2/lib/neo4j/model_schema.rb:131:in `each_schema_element'
	from /usr/local/bundle/gems/neo4j-8.0.2/lib/neo4j/model_schema.rb:94:in `validate_model_schema!'
	from /usr/local/bundle/gems/neo4j-8.0.2/lib/neo4j/active_base.rb:78:in `validate_model_schema!'
	from /usr/local/bundle/gems/neo4j-8.0.2/lib/neo4j/active_base.rb:50:in `new_query'
	from /usr/local/bundle/gems/neo4j-8.0.2/lib/neo4j/active_node/query/query_proxy.rb:305:in `_query'
	from /usr/local/bundle/gems/neo4j-8.0.2/lib/neo4j/active_node/query/query_proxy.rb:290:in `_query_model_as'
	from /usr/local/bundle/gems/neo4j-8.0.2/lib/neo4j/active_node/query/query_proxy.rb:113:in `base_query'
	from /usr/local/bundle/gems/neo4j-8.0.2/lib/neo4j/active_node/query/query_proxy.rb:98:in `query_as'
	from /usr/local/bundle/gems/neo4j-8.0.2/lib/neo4j/active_node/query/query_proxy.rb:87:in `query'
	from /usr/local/bundle/gems/neo4j-8.0.2/lib/neo4j/active_node/query/query_proxy_methods.rb:253:in `first_and_last'
... 12 levels...
	from /usr/local/bundle/gems/activesupport-5.0.1/lib/active_support/dependencies.rb:287:in `load'
	from /usr/local/bundle/gems/activesupport-5.0.1/lib/active_support/dependencies.rb:287:in `block in load'
	from /usr/local/bundle/gems/activesupport-5.0.1/lib/active_support/dependencies.rb:259:in `load_dependency'
	from /usr/local/bundle/gems/activesupport-5.0.1/lib/active_support/dependencies.rb:287:in `load'
	from /usr/local/bundle/gems/spring-2.0.0/lib/spring/commands/rails.rb:6:in `call'
	from /usr/local/bundle/gems/spring-2.0.0/lib/spring/command_wrapper.rb:38:in `call'
	from /usr/local/bundle/gems/spring-2.0.0/lib/spring/application.rb:191:in `block in serve'
	from /usr/local/bundle/gems/spring-2.0.0/lib/spring/application.rb:161:in `fork'
	from /usr/local/bundle/gems/spring-2.0.0/lib/spring/application.rb:161:in `serve'
	from /usr/local/bundle/gems/spring-2.0.0/lib/spring/application.rb:131:in `block in run'
	from /usr/local/bundle/gems/spring-2.0.0/lib/spring/application.rb:125:in `loop'
	from /usr/local/bundle/gems/spring-2.0.0/lib/spring/application.rb:125:in `run'
	from /usr/local/bundle/gems/spring-2.0.0/lib/spring/application/boot.rb:19:in `<top (required)>'
	from /usr/local/lib/ruby/site_ruby/2.3.0/rubygems/core_ext/kernel_require.rb:55:in `require'
	from /usr/local/lib/ruby/site_ruby/2.3.0/rubygems/core_ext/kernel_require.rb:55:in `require'
	from -e:1:in `<main>'
```

Another interesting tidbit is this:

```
railstest_1           | => Booting Puma
railstest_1           | => Rails 5.0.1 application starting in development on http://0.0.0.0:3000
railstest_1           | => Run `rails server -h` for more startup options
railstest_1           | Puma starting in single mode...
railstest_1           | * Version 3.6.2 (ruby 2.3.3-p222), codename: Sleepy Sunday Serenity
railstest_1           | * Min threads: 5, max threads: 5
railstest_1           | * Environment: development
railstest_1           | * Listening on tcp://0.0.0.0:3000
railstest_1           | Use Ctrl-C to stop
railstest_1           | Started GET "/" for 172.19.0.1 at 2017-01-03 23:31:30 +0000
railstest_1           |   ActiveRecord::SchemaMigration Load (2.3ms)  SELECT "schema_migrations".* FROM "schema_migrations"
railstest_1           | Processing by Rails::WelcomeController#index as HTML
...
```

**Comment by cheerfulstoic** [comment]
It's failing trying to setup the logging.  Specifically it's choking because it's (I think) not given a valid session type  How is your app configured for Neo4j in `config/application.rb`, `config/environments/*.rb`, `config/neo4j.yml`, and/or using the NEO4J_URL environment variable?

Also, if you've upgraded from 7.0/6.0 to 8.0/7.0 of `neo4j`/`neo4j-core` you might need to consult this upgrade guide:

http://neo4jrb.readthedocs.io/en/8.0.x/UpgradeGuide.html

**Comment by cheerfulstoic** [comment]
Just released 8.0.5 to make this a bit easier

**Comment by Joshfindit** [comment]
Excellent. Just tested it out, and the error was this:

`ArgumentError (Invalid session type: "server_db" (expected one of [:http, :bolt, :embedded])):`

Changed `config/neo4j.yml` from 

```
  type: server_db
  url: http://neo4j:experimental@neo4j-experimental:7474
```
to
```
  type: http
  url: http://neo4j:experimental@neo4j-experimental:7474
```

(And for kicks, even to `type: bolt`, `url: bolt://neo4j:experimental@neo4j-experimental:7687`)

Initial testing passed (for both `http` and `bolt`).
Hopefully this helps anyone else experiencing the same, and thanks @cheerfulstoic
